src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/bit/domain/Oppfolgingstilfellebit.kt

```diff
@@ -87,16 +87,20 @@ data class OppfolgingstilfelleBit(
 }
 
 fun List<OppfolgingstilfelleBit>.generateOppfolgingstilfelleList(): List<Oppfolgingstilfelle> {
-    val filtrertOppfolgingstilfelleList = this.filtrerOppfolgingstilfelleList()
-    return if (filtrertOppfolgingstilfelleList.isEmpty()) {
+    val filtrertOppfolgingstilfelleBitList = this.filtrerOppfolgingstilfelleBitList()
+    return if (filtrertOppfolgingstilfelleBitList.isEmpty()) {
         emptyList()
     } else {
-        filtrertOppfolgingstilfelleList.toOppfolgingstilfelleDagList()
-            .groupOppfolgingstilfelleList()
+        val oppfolgingstilfelleBiterPerVirksomhet = filtrertOppfolgingstilfelleBitList.groupBy { it.virksomhetsnummer }
+        val oppfolgingstilfelleDagerPerVirksomhet =
+            oppfolgingstilfelleBiterPerVirksomhet.mapValues { it.value.toOppfolgingstilfelleDagList() }
+
+        filtrertOppfolgingstilfelleBitList.toOppfolgingstilfelleDagList()
+            .groupOppfolgingstilfelleList(oppfolgingstilfelleDagerPerVirksomhet)
     }
 }
 
-private fun List<OppfolgingstilfelleBit>.filtrerOppfolgingstilfelleList(): List<OppfolgingstilfelleBit> {
+private fun List<OppfolgingstilfelleBit>.filtrerOppfolgingstilfelleBitList(): List<OppfolgingstilfelleBit> {
```

Comment:
Kan metodenavnet her si noe om hva som er kriteriet for filtrering?

Reply:
Kan se på det